json-file-fix.mjs

```diff
@@ -0,0 +1,76 @@
+import fs from 'fs';
+import path from 'path';
+
+// This script does a number of things:
+// - Normalises the descriptions and solutions to be "Unknown cause." and "Unknown solution." rather than "N/A"
+// - If one of the descriptions or solutions is "Unknown cause." or "Unknown solution.", the other will be copied to it
+// - Ensures all descriptions and solutions end with a full stop if they do not already and if the last character is a letter
+// - Writes the changes back to the file in place
+
+const arg = process.argv[2];
+if (!arg || !fs.existsSync(arg) || !fs.lstatSync(arg).isDirectory()) {
+	console.log("Please provide a valid directory path.");
+	process.exit(1);
+}
+
+fs.readdirSync(arg, { recursive: true, withFileTypes: true }).forEach(file => {
+	if (!file.isFile() && !file.name.endsWith(".json")) return;
+
+	const fullPath = path.join(file.parentPath, file.name);
+
+	console.log(`Processing ${fullPath}`);
+
+	const data = fs.readFileSync(fullPath, 'utf8');
+	const json = JSON.parse(data);
+
+	const key1 = Object.keys(json)[0];
+	const key2 = Object.keys(json[key1])[0];
+
+	const errorObject = json[key1][key2];
+	if (!errorObject || typeof errorObject !== "object") {
+		console.log(`No error object found in ${fullPath}`);
+		return;
+	}
+
+	if (!errorObject["short_description"] || !errorObject["long_description"] || !errorObject["short_solution"] || !errorObject["long_solution"]) {
+		console.log(`Missing fields in ${fullPath}`);
+		return;
+	}
+
+	if (errorObject["short_description"] === "N/A") errorObject["short_description"] = "Unknown cause.";
+	if (errorObject["long_description"] === "N/A") errorObject["long_description"] = "Unknown cause.";
+	if (errorObject["short_solution"] === "N/A") errorObject["short_solution"] = "Unknown solution.";
+	if (errorObject["long_solution"] === "N/A") errorObject["long_solution"] = "Unknown solution.";
```

Comment:
These may catch some intended areas. I just did a quick check and there's only 4 uses of `N/A` right now (as well as some uses of `Not applicable` for some reason, idk why this is inconsistent)

2 of these `N/A` instances are for "success errors", so they really *are* N/A and having them say `Unknown cause/solution` may be confusing

Though idk what the best way to handle this would be. We could manually fix those cases to be `N/A` again but then any subsequent runs of this script would change them again

Reply:
I'm happy to remove it - I thought it was a bit hacky myself anyway tbf

Reply:
That also works if you think it's fine :+1: